pkg/service/manifest/service.go

```diff
@@ -286,13 +330,93 @@ func (s Service) ProcessApplicationSubmission(ctx context.Context, request model
 		return nil, util.LoggingErrorMsg(err, "could not store application")
 	}
 
-	storedOp := opstorage.StoredOperation{
+	storedOp := &opstorage.StoredOperation{
 		ID: opID,
 	}
-	if err = s.opsStorage.StoreOperation(ctx, storedOp); err != nil {
+	if err = s.opsStorage.StoreOperation(ctx, *storedOp); err != nil {
 		return nil, errors.Wrap(err, "storing operation")
 	}
-	return operation.ServiceModel(storedOp)
+
+	autoStoredOp, err := s.maybeIssueAutomatically(ctx, request, manifestID, applicantDID, applicationID, gotManifest)
+	if err != nil {
+		return nil, err
+	}
+
+	if autoStoredOp != nil {
+		storedOp = autoStoredOp
+	}
+	return operation.ServiceModel(*storedOp)
+}
+
+func (s Service) maybeIssueAutomatically(
+	ctx context.Context,
+	request model.SubmitApplicationRequest,
+	manifestID string,
+	applicantDID string,
+	applicationID string,
+	gotManifest *manifeststg.StoredManifest,
+) (*opstorage.StoredOperation, error) {
+	issuanceTemplates, err := s.issuanceTemplateStorage.GetIssuanceTemplatesByManifestID(ctx, manifestID)
+	if err != nil {
+		return nil, errors.Wrap(err, "fetching issuance templates by manifest ID")
+	}
+	if len(issuanceTemplates) == 0 {
+		return nil, nil
+	}
+
+	issuanceTemplate := issuanceTemplates[0].IssuanceTemplate
+
+	if len(issuanceTemplates) > 1 {
```

Comment:
maybe turn this into two checks

`if len(issuanceTemplates) > 0)` and then > 1

Reply:
disregard this